Src/BootCamp.Chapter/Checks.cs

```diff
@@ -1,39 +1,214 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Text;
+﻿using System.Text;
 
 namespace BootCamp.Chapter
 {
     // This class is used to have a freedom of design, but with tests applied.
+
     public static class Checks
     {
+        const string nA = "N/A.";
         public static string FindMostPoorPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (Testers.IsThisStringArrayValid(peopleAndBalances))
+            {
+                return nA;
+            }
+            StringBuilder name = new StringBuilder();
+            decimal amount = 0;
+            int personCount = 1;
+
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                IsMorePoor(AccountCreator(peopleAndBalances[i]), i);
+            }
+
+            return MakeStuffWithInput.MakefullstringPoorestPeople(personCount, name.ToString(), amount);
+
+            void IsMorePoor(Account account, int i)
+            {
+                if (i == 0) // first account to be added is alway's the poorest.
+                {
+                    name.Append(account.GetName());
+                    amount = account.CurrentBalance();
+                }
+                else if (amount == account.CurrentBalance()) // this account has the same balance as current.
+                {
+                    if (personCount == 1) // the second account to be added
+                    {
+                        name.Append(" and " + account.GetName());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Define a constant instead of using this literal ' and ' 12 times.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5123803)